src/core/dfn-finder.js

```diff
@@ -174,11 +176,18 @@ function getDfns(dfnForArray, parent, originalName, type) {
   }
   // Definitions that have a name and [data-dfn-for] that exactly match the
   // IDL entity:
-  const dfns = dfnForArray.filter(dfn => dfn.dataset.dfnFor === parent);
-  // If this is a top-level entity, and we didn't find anything with
-  // an explicitly empty [for], try <dfn> that inherited a [for].
+  const dfns = [];
+  for (const dfn of dfnForArray) {
+    /** @type {HTMLElement} */
+    const closestDfnFor = dfn.closest(`[data-dfn-for]`);
+    if (!closestDfnFor) {
+      continue;
+    }
+    if (closestDfnFor.dataset.dfnFor === parent) dfns.push(dfn);
+  }
   if (dfns.length === 0 && parent === "" && dfnForArray.length === 1) {
-    return dfnForArray;
+    // Make sure the name exactly matches
+    return dfnForArray[0].textContent === originalName ? dfnForArray : [];
```

Comment:
Is this just to make sure or did you see some mismatches?

Reply:
I seem to recall getting mismatches.